### PR TITLE
Fix NormalizerAgg test searcher wrapping

### LINKED ISSUE #57164: test failure: org.elasticsearch.xpack.analytics.normalize.NormalizeAggregatorTests.testTermsAggParent
**Build scan**:

https://gradle-enterprise.elastic.co/s/zb66vt4jdhzgk

**Repro line**:

```
./gradlew ':x-pack:plugin:analytics:test' --tests "org.elasticsearch.xpack.analytics.normalize.NormalizeAggregatorTests.testTermsAggParent" -Dtests.seed=921D46F82701E024 -Dtests.security.manager=true -Dtests.locale=sq -Dtests.timezone=America/Yellowknife -Dcompiler.java=14 -Druntime.java=11
```
**Reproduces locally?**:

Yes.

**Applicable branches**:

7.x 

**Failure history**:

Recently added feature: https://github.com/elastic/elasticsearch/pull/56399 cc: @talevy 

**Failure excerpt**:

```
org.elasticsearch.xpack.analytics.normalize.NormalizeAggregatorTests > testTermsAggParent FAILED
    java.lang.ClassCastException: class org.apache.lucene.index.ParallelCompositeReader cannot be cast to class org.apache.lucene.index.DirectoryReader (org.apache.lucene.index.ParallelCompositeReader and org.apache.lucene.index.DirectoryReader are in unnamed module of loader 'app')
        at __randomizedtesting.SeedInfo.seed([921D46F82701E024:928D3EDD3E4FD870]:0)
        at org.elasticsearch.search.aggregations.support.ValuesSource$Bytes$WithOrdinals$FieldData.globalOrdinalsValues(ValuesSource.java:206)
        at org.elasticsearch.search.aggregations.support.ValuesSource$Bytes$WithOrdinals.globalMaxOrd(ValuesSource.java:179)
        at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorFactory.getMaxOrd(TermsAggregatorFactory.java:300)
        at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorFactory.access$100(TermsAggregatorFactory.java:52)
        at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorFactory$1.build(TermsAggregatorFactory.java:108)
        at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorFactory.doCreateInternal(TermsAggregatorFactory.java:274)
        at org.elasticsearch.search.aggregations.support.ValuesSourceAggregatorFactory.createInternal(ValuesSourceAggregatorFactory.java:49)
        at org.elasticsearch.search.aggregations.AggregatorFactory.create(AggregatorFactory.java:250)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.createAggregator(AggregatorTestCase.java:244)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.createAggregator(AggregatorTestCase.java:222)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.searchAndReduce(AggregatorTestCase.java:493)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.searchAndReduce(AggregatorTestCase.java:441)
        at org.elasticsearch.xpack.analytics.normalize.NormalizeAggregatorTests.testCase(NormalizeAggregatorTests.java:165)
        at org.elasticsearch.xpack.analytics.normalize.NormalizeAggregatorTests.testTermsAggParent(NormalizeAggregatorTests.java:120)
```

### PR DESCRIPTION
**Pull request by talevy**
The searcher was randomly wrapping its reader as slow, parallel, or filtered.
This was causing casting issues in the normalizer tests. By removing the
wrapping, the problem goes away.

Closes #57164

### CODE REVIEW

File: x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/normalize/NormalizeAggregatorTests.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ private void testCase(ValuesSourceAggregationBuilder<?> aggBuilder, Consumer<Int
             termFieldType.setHasDocValues(true);
 
             try (IndexReader indexReader = DirectoryReader.open(directory)) {
-                IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
```
Comment (nik9000) [inline x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/normalize/NormalizeAggregatorTests.java:163]: @polyfractal just pointed me to `AggregatorTestCase#newIndexSearcher` which seems to do that job. It has even less randomization, but it looks to be what we're moving the tests to.

Comment (talevy) [reply x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/normalize/NormalizeAggregatorTests.java:163]: what is the reasoning behind maybeWrapping? and why is it unsupported in some scenarios?

Comment (talevy) [reply x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/normalize/NormalizeAggregatorTests.java:163]: and thank you for the heads up. I've updated to use the new method

**Review by nik9000 — APPROVED** [review]
LGTM.

If you mean `boolean maybeWrap = unwrapped == indexReader;` from `AggregatorTestCase`, I'm removing it now. It prevented us from using lucene's wrappers when we needed global ordinals. But now that we're using `newIndexSearcher` we won't use the wrapper anyway.

**Pull request merged**